### PR TITLE
ci: enable cra-scan

### PR DESCRIPTION
**Pull request by iamar7**
### Description

Enable CRA scan for terraform-ibm-landing-zone

### Types of changes in this PR

https://github.ibm.com/GoldenEye/issues/issues/4112

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [x] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,3 +15,6 @@ jobs:
   call-terraform-ci-pipeline:
     uses: terraform-ibm-modules/common-pipeline-assets/.github/workflows/common-terraform-module-ci.yml@v1.7.3
     secrets: inherit
+    with:
+      craTarget: "examples/basic"
```
Comment (ocofaigh) [inline .github/workflows/ci.yml:19]: `examples/basic` is probably the wrong directory to scan here. We should probably be scanning `patterns/roks`. We may also need to scan `patterns/vsi` but I don't think our pipeline currently supports scanning multiple directories, so maybe go with roks for now

Comment (iamar7) [reply .github/workflows/ci.yml:19]: Made the required changes

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,3 +15,6 @@ jobs:
   call-terraform-ci-pipeline:
     uses: terraform-ibm-modules/common-pipeline-assets/.github/workflows/common-terraform-module-ci.yml@v1.7.3
     secrets: inherit
+    with:
+      craTarget: "patterns/roks"
+      craGoalIgnoreFile: "cra-tf-validate-ignore-goals.json"
```
Comment (ocofaigh) [inline .github/workflows/ci.yml:20]: please remove the default values you added in patterns/roks/variables.tf and instead pass them as TF_VARs here:

`craEnvironmentVariables: "TF_VAR_prefix=land-zone-roks,TF_VAR_region=us-south"`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: patterns/roks/variables.tf [hunk 3]
```diff
@@ -11,16 +11,18 @@ variable "ibmcloud_api_key" {
 variable "prefix" {
   description = "A unique identifier for resources. Must begin with a lowercase letter and end with a lowercase letter or number. This prefix will be prepended to any resources provisioned by this template. Prefixes must be 16 or fewer characters."
   type        = string
+  default     = "land-zone-roks"
 
   validation {
     error_message = "Prefix must begin with a letter and contain only lowercase letters, numbers, and - characters. Prefixes must end with a lowercase letter or number and be 16 or fewer characters."
-    condition     = can(regex("^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$", var.prefix)) && length(var.prefix) <= 13
+    condition     = can(regex("^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$", var.prefix)) && length(var.prefix) <= 16
```
[the remark this answers is not in the document]
Comment (iamar7) [reply patterns/roks/variables.tf:18]: Changed it back to 13.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (iamar7) [reply .github/workflows/ci.yml:20]: Done

**Comment by iamar7** [comment]
> see comments

Made the changes accordingly

File: cra-tf-validate-ignore-goals.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            "is_valid": false
+        },
+        {
+            "scc_goal_id": "3000452",
+            "description": "Check whether Virtual Private Cloud (VPC) network access control lists don't allow egress from 0.0.0.0/0 to any port",
+            "ignore_reason": "This is part of the test example and not the current module",
+            "is_valid": false
+        }
+    ]
+}
```
Comment (ocofaigh) [inline cra-tf-validate-ignore-goals.json:40]: What you have said in the ingore_reason is not true. The roks pattern deploys COS and VPC so all of these are valid results. I'll need to investigate each of them to decide what we do, so leave this with me

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**